### PR TITLE
Fix some memory leaks

### PR DESCRIPTION
**Pull request by clumens**

### CODE REVIEW

File: lib/pengine/tests/rules/pe_cron_range_satisfied_test.c [hunk 1]
```diff
@@ -11,7 +11,7 @@ run_one_test(const char *t, const char *x, int expected) {
     g_assert_cmpint(pe_cron_range_satisfied(tm, xml), ==, expected);
 
     crm_time_free(tm);
-    xmlFreeNode(xml);
+    xmlFreeDoc(xml->doc);
```
Comment (kgaillot) [inline lib/pengine/tests/rules/pe_cron_range_satisfied_test.c:14]: this is fine, but fyi free_xml() handles this generally

we should add a copyright/license notice to this file

**Review by kgaillot — COMMENTED** [review]
Looks good. Not sure why the cli regression tests are different

**Comment by clumens** [comment]
> Looks good. Not sure why the cli regression tests are different

Something's wrong with the pcmk__cmdline_preproc change.  Working on it.

**Comment by clumens** [comment]
Okay, -fsanitize=asan is all about the runtime checks so we're going to have to run everything if we want to fix everything it finds.  I've reduced a run of `cts/cts-cli -s -v` down to 26 test failures, all of which seem to be one of these two:
```
Indirect leak of 96 byte(s) in 4 object(s) allocated from:
    #0 0x7f33a8591d18 in __interceptor_malloc (/lib64/libasan.so.5+0x10dd18)
    #1 0x7f33a7bdec98 in g_malloc (/lib64/libglib-2.0.so.0+0x55c98)
```

```
Direct leak of 264 byte(s) in 3 object(s) allocated from:
    #0 0x7f51efab8d18 in __interceptor_malloc (/lib64/libasan.so.5+0x10dd18)
    #1 0x7f51ef105c98 in g_malloc (/lib64/libglib-2.0.so.0+0x55c98)

Direct leak of 120 byte(s) in 3 object(s) allocated from:
    #0 0x7f51efab8f16 in __interceptor_calloc (/lib64/libasan.so.5+0x10df16)
    #1 0x7f51ef4ca206 in new_private_data /home/clumens/src/pacemaker/lib/common/xml.c:250

Direct leak of 36 byte(s) in 3 object(s) allocated from:
    #0 0x7f51efab8d18 in __interceptor_malloc (/lib64/libasan.so.5+0x10dd18)
    #1 0x7f51eeff1bd5 in xmlStrndup (/lib64/libxml2.so.2+0xacbd5)

Indirect leak of 9240 byte(s) in 77 object(s) allocated from:
    #0 0x7f51efab8d18 in __interceptor_malloc (/lib64/libasan.so.5+0x10dd18)
    #1 0x7f51eefa05cd  (/lib64/libxml2.so.2+0x5b5cd)

Indirect leak of 4800 byte(s) in 50 object(s) allocated from:
    #0 0x7f51efab8d18 in __interceptor_malloc (/lib64/libasan.so.5+0x10dd18)
    #1 0x7f51eef9f31e in xmlNewDocProp (/lib64/libxml2.so.2+0x5a31e)

Indirect leak of 3920 byte(s) in 98 object(s) allocated from:
    #0 0x7f51efab8f16 in __interceptor_calloc (/lib64/libasan.so.5+0x10df16)
    #1 0x7f51ef4ca206 in new_private_data /home/clumens/src/pacemaker/lib/common/xml.c:250

Indirect leak of 1420 byte(s) in 157 object(s) allocated from:
    #0 0x7f51efab8d18 in __interceptor_malloc (/lib64/libasan.so.5+0x10dd18)
    #1 0x7f51eeff1bd5 in xmlStrndup (/lib64/libxml2.so.2+0xacbd5)

Indirect leak of 1080 byte(s) in 9 object(s) allocated from:
    #0 0x7f51efab8d18 in __interceptor_malloc (/lib64/libasan.so.5+0x10dd18)
    #1 0x7f51eef9b4db in xmlNewText (/lib64/libxml2.so.2+0x564db)

Indirect leak of 880 byte(s) in 5 object(s) allocated from:
    #0 0x7f51efab8d18 in __interceptor_malloc (/lib64/libasan.so.5+0x10dd18)
    #1 0x7f51eef9ac59 in xmlNewDoc (/lib64/libxml2.so.2+0x55c59)

Indirect leak of 864 byte(s) in 9 object(s) allocated from:
    #0 0x7f51efab8d18 in __interceptor_malloc (/lib64/libasan.so.5+0x10dd18)
    #1 0x7f51eef9b5cf  (/lib64/libxml2.so.2+0x565cf)

Indirect leak of 840 byte(s) in 7 object(s) allocated from:
    #0 0x7f51efab8d18 in __interceptor_malloc (/lib64/libasan.so.5+0x10dd18)
    #1 0x7f51eef9b274 in xmlNewNode (/lib64/libxml2.so.2+0x56274)

Indirect leak of 704 byte(s) in 5 object(s) allocated from:
    #0 0x7f51efab914e in realloc (/lib64/libasan.so.5+0x10e14e)
    #1 0x7f51ef105d4f in g_realloc (/lib64/libglib-2.0.so.0+0x55d4f)

Indirect leak of 509 byte(s) in 28 object(s) allocated from:
    #0 0x7f51efa4152d in strdup (/lib64/libasan.so.5+0x9652d)
    #1 0x7f51ef4a1d8b in cluster_option_value /home/clumens/src/pacemaker/lib/common/options.c:520

Indirect leak of 133 byte(s) in 28 object(s) allocated from:
    #0 0x7f51efa4152d in strdup (/lib64/libasan.so.5+0x9652d)
    #1 0x7f51ef4a1d80 in cluster_option_value /home/clumens/src/pacemaker/lib/common/options.c:519

Indirect leak of 64 byte(s) in 2 object(s) allocated from:
    #0 0x7f51efab8f16 in __interceptor_calloc (/lib64/libasan.so.5+0x10df16)
    #1 0x7f51ef105cf0 in g_malloc0 (/lib64/libglib-2.0.so.0+0x55cf0)

Indirect leak of 31 byte(s) in 2 object(s) allocated from:
    #0 0x7f51efa4152d in strdup (/lib64/libasan.so.5+0x9652d)
    #1 0x7f51ef99355a in populate_hash /home/clumens/src/pacemaker/lib/pengine/rules.c:398
    #2 0x7f51ef99355a in unpack_attr_set /home/clumens/src/pacemaker/lib/pengine/rules.c:500
    #3 0x7f51ef99355a in unpack_attr_set /home/clumens/src/pacemaker/lib/pengine/rules.c:478

Indirect leak of 11 byte(s) in 2 object(s) allocated from:
    #0 0x7f51efa4152d in strdup (/lib64/libasan.so.5+0x9652d)
    #1 0x7f51ef99354f in populate_hash /home/clumens/src/pacemaker/lib/pengine/rules.c:398
    #2 0x7f51ef99354f in unpack_attr_set /home/clumens/src/pacemaker/lib/pengine/rules.c:500
    #3 0x7f51ef99354f in unpack_attr_set /home/clumens/src/pacemaker/lib/pengine/rules.c:478
```

The former is probably just stuff we ignore via valgrind suppressions.  I can't track it down.  There's not really enough information to know where that's happening.  There's several other things going on in the latter.  First, there's several things happening relating to xml functions.  I think that's because `free_private_data` doesn't actually free private data always.  And then second, there's a couple hash table related functions that are probably due to us not registering a free function somewhere.  But we use hash tables all over the place.  It'd be nice if it would give us longer stack traces.

**Pull request merged**